### PR TITLE
Add 3d smart camera

### PR DESCRIPTION
**Pull request by dNechita**

### CODE REVIEW

File: sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp [hunk 1]
```diff
@@ -0,0 +1,599 @@
...
+    status = m_depthSensor->getHandle(&handle);
+    if (status != Status::OK) {
+        LOG(ERROR) << "Failed to obtain the handle";
+        return status;
+    }
+
+    m_details.bitCount = 12;
+
+    // Open communication with the RGB sensor
+    status = m_depthSensor->open();
```
Comment (SeptimiuVana) [inline sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp:151]: Shouldn't it be m_rgbSensor here?

File: sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp [hunk 2]
```diff
@@ -0,0 +1,599 @@
...
+    Status status = Status::OK;
+
+    details = m_details;
+
+    return status;
+}
+
+std::shared_ptr<aditof::DepthSensorInterface> Camera3D_Smart::getSensor() {
+    return m_depthSensor;
+}
```
Comment (SeptimiuVana) [inline sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp:459]: Can we add both sensors here? or we will keep it only for the depthSensor?

File: sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp (hunk 1 above)
Comment (dNechita) [reply sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp:151]: yes! thanks!

File: sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp (hunk 2 above)
Comment (dNechita) [reply sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp:459]: Yes we should return both. I will have to change the API and make changes for all the other cameras.

File: sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp [hunk 3]
```diff
@@ -0,0 +1,599 @@
...
+    if (!depthSensor) {
+        LOG(WARNING) << "Invalid instance of a depth sensor";
+        return;
+    }
+    aditof::SensorDetails sDetails;
+    m_depthSensor->getDetails(sDetails);
+    m_details.connection = sDetails.connectionType;
+
+    // Check RGB Sensor
+    if (!depthSensor) {
```
Comment (SeptimiuVana) [inline sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp:74]: Shouldn't is be rgbSensor?

File: sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp (hunk 1 above)
Comment (dNechita) [reply sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp]: fixed

File: sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp (hunk 2 above)
Comment (dNechita) [reply sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp:464]: fixed

File: sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp (hunk 3 above)
Comment (dNechita) [reply sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp]: fixed

**Review by SeptimiuVana — APPROVED** [review]

**Pull request merged**